### ISSUE TITLE
inference inconsistent after inheritance, not-callable false positive

### ISSUE THREAD

**Issue by alvarocaceres** [PyCQA/pylint#1213]
### Steps to reproduce
```
import foo  # Note: pylint doesn't get access to foo

class D(foo.Blah):
  pass

a = foo.Blah()  # instance of a class that defines __call__
a() # no error

d = D()
d() # not-callable error
```
### Current behavior

`not-callable error` 

### Expected behavior

`d()` should not generate a not-callable error

### pylint --version output
```
$ pylint --version
No config file found, using default configuration
pylint 1.6.4, 
astroid 1.4.8
Python 3.4.3 (default, Nov 17 2016, 01:08:31) 
[GCC 4.8.4]
```

**Comment by PCManticore** [comment]
Yes, I can reproduce this.
